components/ws-manager-mk2/controllers/status.go

```diff
@@ -268,12 +266,13 @@ func (r *WorkspaceReconciler) extractFailure(ctx context.Context, ws *workspacev
 			// we would not be here as we've checked for a DeletionTimestamp prior. So let's find out why the
 			// container is terminating.
 			if terminationState.ExitCode != 0 && terminationState.Message != "" {
-				var phase workspacev1.WorkspacePhase
+				var phase *workspacev1.WorkspacePhase
```

Comment:
without the pointer, this was defaulting to the empty string, and since that's non-nil it will get set as the status in the calling function. Similar to elsewhere in this function, this should be a pointer and default to nil